common/containers/Tabs/SendTransaction/index.jsx

```diff
@@ -238,8 +267,63 @@ export class SendTransaction extends React.Component {
     return { to, data, value, unit, gasLimit, readOnly };
   }
 
+  isValid() {
+    const { to, value } = this.state;
+    return (
+      isValidETHAddress(to) &&
+      value &&
+      Number(value) > 0 &&
+      !isNaN(Number(value)) &&
+      isFinite(Number(value))
+    );
+  }
+
+  // FIXME MOVE ME
+  getTransactionFromState() {
+    // FIXME add gas price
+    if (this.state.unit === 'ether') {
+      return {
+        to: this.state.to,
+        from: this.props.wallet.getAddress(),
+        // gasPrice: `0x${new Number(50 * 1000000000).toString(16)}`,
+        value: '0x' + toHex(toWei(Big(this.state.value), 'ether'))
+      };
+    }
+    const token = this.props.tokens.find(x => x.symbol === this.state.unit);
+    if (!token) {
+      return;
+    }
+
+    return {
+      to: token.address,
+      from: this.props.wallet.getAddress(),
+      // gasPrice: `0x${new Number(50 * 1000000000).toString(16)}`,
+      value: '0x0',
+      data: ERC20.transfer(
+        this.state.to,
+        Big(this.state.value).times(Big(10).pow(token.decimal))
+      )
+    };
+  }
+
+  estimateGas(state: State) {
```

Comment:
The only thing I think needs moving is the gasLimitString stuff. Otherwise the logic in here about state change should stay in the component, since it's component-centric.